cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -85,7 +85,12 @@ public static void runProgram(Path sourceRootPath, Path sourcePath, boolean runS
                 !RepoUtils.hasProjectRepo(sourceRootPath)) {
             programFile = compile(fullPath.getParent(), fullPath.getFileName(), offline);
         } else if (Files.isDirectory(sourceRootPath)) {
-            programFile = compile(sourceRootPath, sourcePath, offline);
+            if (RepoUtils.hasProjectRepo(sourceRootPath)) {
```

Comment:
is this correct. isn't sourceRootPath always a dir. shouldn't this check be on sourcePath?

Reply:
Fixed with 424ab3a